system/jlib/jsocket.cpp

```diff
@@ -2125,15 +2125,18 @@ bool CSocket::check_connection()
         rc = sendto(sock, NULL, 0, 0, &u.sa, ul);
     }
     else {
-        rc = send(sock, NULL, 0, SEND_FLAGS);
+        char buffer[4];
+        rc = recv(sock, buffer, sizeof(buffer), MSG_PEEK | MSG_DONTWAIT);
```

Comment:
I don't think MSG_DONTWAIT is implemented in windows.

Reply:
It might ork if you set non blocking mode, and then read, but what happens if something else is using the socket at the same time?  Do we protect against that?

Reply:
As I have no way to test what happens in Windows (and it doesn't really matter), I reverted the windows version of the code to how it was